src/app/modules/shared/services/shopping.cart.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { OrderProduct } from '../types/product.order';
+
+@Injectable({ providedIn: 'root' })
+export class ShoppingCartService {
+  private shoppingCart: OrderProduct[] = [];
+  public getShoppingCart() {
+    return this.shoppingCart;
+  }
+
+  public addToShoppingCart(product: OrderProduct): OrderProduct[] {
+    const foundProduct = this.shoppingCart.find((p) => p.id === product.id);
+    if (!foundProduct) {
+      product.quantity = 0;
+      product.quantity++;
```

Comment:
Here wouldn't be equal if you directly set the product quantity to 1? I see you set it to 0 and then increment. I don't find a reason for breaking the operation in these to 2 statements.